pkg/controllers/remediation/eventhandlers_test.go

```diff
@@ -0,0 +1,372 @@
+				operation: "Delete",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+				},
+			},
+			wantQLen: 0,
+		},
+		{
+			name: "update event: equal cluster condition",
+			args: args{
+				operation: "Update",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionFalse,
+							},
+						},
+					},
+				},
+				oldObj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionFalse,
+							},
+						},
+					},
+				},
+			},
+			wantQLen: 0,
+		},
+		{
+			name: "update event: not equal cluster condition",
+			args: args{
+				operation: "Update",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionFalse,
+							},
+						},
+					},
+				},
+				oldObj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionTrue,
+							},
+						},
+					},
+				},
+			},
+			wantQLen: 1,
+		},
+		{
+			name: "generic event",
+			args: args{
+				operation: "Generic",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+				},
+			},
+			wantQLen: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			queue := tt.args.q
+			h := newClusterEventHandler()
+			switch tt.args.operation {
+			case "Create":
+				createEvent := event.CreateEvent{Object: tt.args.obj}
+				h.Create(context.TODO(), createEvent, queue)
+			case "Delete":
+				deleteEvent := event.DeleteEvent{Object: tt.args.obj}
+				h.Delete(context.TODO(), deleteEvent, queue)
+			case "Update":
+				updateEvent := event.UpdateEvent{ObjectNew: tt.args.obj, ObjectOld: tt.args.oldObj}
+				h.Update(context.TODO(), updateEvent, queue)
+			case "Generic":
+				genericEvent := event.GenericEvent{Object: tt.args.obj}
+				h.Generic(context.TODO(), genericEvent, queue)
+			default:
+				t.Errorf("no support operation %v", tt.args.operation)
+				return
+			}
+
+			if got := queue.Len(); got != tt.wantQLen {
+				t.Errorf("clusterEventHandler Create() process queue length = %v, want %v", got, tt.wantQLen)
```

Comment:
This log seems incorrect, as here not just focus on `Create`.

Reply:
updated, thx~